### PR TITLE
bpo-38580: Document that select() accepts iterables, not just sequences

### PR DESCRIPTION
**Pull request by jstasiak**
~~I think this one can go through without an issue number.~~

https://bugs.python.org/issue38580

### CODE REVIEW

**Review by brandtbucher — APPROVED** [review]

**Comment by brandtbucher** [comment]
The argument clinic docstring for `select_select_impl` in `Modules/selectmodule.c` can probably be updated too... but I think this PR is fine without it.

**Review by corona10 — COMMENTED** [review]
As @brandtbucher review,
Please update the docstring of https://github.com/python/cpython/blob/303475e87372aacb17990802538e440735a69525/Modules/selectmodule.c#L235

**Comment by jstasiak** [comment]
> As @brandtbucher review,
> Please update the docstring of
> 
> https://github.com/python/cpython/blob/303475e87372aacb17990802538e440735a69525/Modules/selectmodule.c#L235

Done.

**Review by corona10 — COMMENTED** [review]
@jstasiak cc @brandtbucher 

IMHO, I don't know this is the right patch.
I am not a select module expert, but if this patch is accepted.
Docstring means that it will accept the dictionary for the  `rlist, wlist, xlist`.
(I am not meaning select.select({}, {}, {}, 1) rasing TypeError).
But is this working properly?

**Comment by brandtbucher** [comment]
@corona10 Yes, it should be fine. This function calls `seq2set` on each iterable, which uses `PySequence_Fast` to convert them to lists (or similar) before doing any work.

Any finite iterable should be okay here.

**Comment by jstasiak** [comment]
> Docstring means that it will accept the dictionary for the `rlist, wlist, xlist`.

@corona10 Yeah it accepts dictionaries now, that's how I discovered that the documentation is being overly strict here.

**Review by corona10 — CHANGES_REQUESTED** [review]
@jstasiak 
> Yeah it accepts dictionaries now, that's how I discovered that the documentation is being overly strict here.

I think that we should discuss with core developers which part should be changed.

- Updating implementation to only accept sequence type and raising TypeError
or
- Updating docs.

I 'd like to recommend it to open a new issue on bug.python.org
After the decision is finalized we can finish this PR.

So I'm left request change, not to this PR but request to opening the issue.

Thanks!

**Comment by brandtbucher** [comment]
Okay then. Since this more controversial than I anticipated, I think a BPO issue should be opened for this.

**Comment by jstasiak** [comment]
That's fair, just opened https://bugs.python.org/issue38580

**Comment by jstasiak** [comment]
There's no interest in [the issue](https://bugs.python.org/issue38580) so far, so I let myself mention two recent select module contributors in hope I receive some valuable input here: @vstinner @taleinat

File: Modules/selectmodule.c [hunk 1]
```diff
@@ -232,11 +232,11 @@ select.select
 
 Wait until one or more file descriptors are ready for some kind of I/O.
 
-The first three arguments are sequences of file descriptors to be waited for:
+The first three arguments are iterables of file descriptors to be waited for:
 rlist -- wait until ready for reading
 wlist -- wait until ready for writing
 xlist -- wait for an "exceptional condition"
-If only one kind of condition is required, pass [] for the other lists.
```
Comment (taleinat) [inline Modules/selectmodule.c]: In this line, the original "pass []" is clearer and simpler than "pass empty iterables", and is still correct.

**Review by taleinat — CHANGES_REQUESTED** [review]
Thanks for following this up, @jstasiak. This looks very good to me, I have just one minor comment.

File: Modules/selectmodule.c (hunk 1 above)
Comment (jstasiak) [reply Modules/selectmodule.c]: That's fair, I brought back the original.

**Comment by jstasiak** [comment]
@taleinat Should I write a news entry or is it fine as it is?

**Comment by taleinat** [comment]
> @taleinat Should I write a news entry or is it fine as it is?

Since this is just updating the documentation to better describe the existing behavior, IMO there's no need for a NEWS entry, as for other normal doc-only changes.

**Pull request merged**